js/packages/teams-ai/src/AIApiFactory.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * @module teams-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TurnState } from './TurnState';
+import { DefaultTurnState } from './DefaultTurnStateManager';
+import { PromptTemplate } from './Prompts';
+import { ConfiguredAIOptions } from './AI';
+import { TurnContext } from 'botbuilder';
+import { Plan } from './Planner';
+import { ResponseParser } from './ResponseParser';
+
+export interface AIApiFactoryOptions {
+    /**
+     * Optional. A flag indicating if the planner should only say one thing per turn.
+     * @summary
+     * The planner will attempt to combine multiple SAY commands into a single SAY command when true.
+     * Defaults to false.
+     */
+    oneSayPerTurn?: boolean;
+}
+
+export interface AIApiFactory<TState extends TurnState = DefaultTurnState> {
```

Comment:
Thoughts on naming this `LLMPlanner`, since it's identical to the planner interface, but specialized for LLMs i.e prompts

Reply:
Are we planning on providing functionality outside of the chat domain that is (for example) `createImage` instead of `createPlan`? Unless we are, I'm not sure I understand the benefit of renaming this. Could you elaborate? Conceptually, LLMPlanner seems to apply more to SK than our library, since it implies the existence of other types of Planners.